### PR TITLE
Web6102 distance helper

### PR DESCRIPTION
**Pull request by MCooke**
#### What are the links to relevant tickets?
https://hxshortbreaks.atlassian.net/browse/WEB-6102
#### What does this PR do?
Adds in helper for the wall to calc distance from two locations using lat and long.
#### What functional/unit tests does this PR have?
0
#### How should a developer review this?
Eyes over helper, come have a chat.
#### Screenshots (if appropriate)

screen shot 2015-10-01 at 16 58 07

---
#### Quality checklist (for PR author to complete BEFORE code review):
- [x] I've checked there is appropriate unit test coverage.
- [x] I've updated documentation with any changes to procedures.
- [x] I've checked this work against the requirements of the Jira.
- [x] This change passes all necessary tests (cross browser, automated or otherwise).
- [x] I am ready for this to be code reviewed, merged and tested.

#### Reviewer 1 @lukefrake
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

#### Reviewer 2 @t1gr0u
- [x] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

File: helpers/calcDistance.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  * @param {number} lat1 latitude of first position
+  * @param {number} lon1 longitude of first position
+  * @param {number} lat2 latitude of second position
+  * @param {number} lon2 longitude of second position
+  * @param {string} unit defaults to M (Miles), accepts K (Kilometers) and N (nautical miles)
+  * @param {number} decimalPlaces how many decimal places to return
+  * @example {@_calcDistance lat1=50.0000 lon1=0.12000 lat2=51.0000 lon2=0.11000- decimalPlaces=2} {/_calcDistance} output template distance
+  */
+
+  dust.helpers._calcDistance = function( chunk, context, bodies, params ) {
```
Comment (lukefrake) [inline helpers/calcDistance.js]: These distances are as the crow flies.

We've always had problems displaying distance in this way as it's much closer than actual driving distance which is what people expect to see!

Comment (lukefrake) [inline helpers/calcDistance.js]: This is perfect though, as long as thats what you expect :dancer: :dancers:

Comment (MCooke) [inline helpers/calcDistance.js]: No, i expect 7 lines of JS to calc distance as a car while making toast.

File: helpers/calcDistance.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    var lon1 = dust.helpers.tap( params.lon1, chunk, context );
+    var lat2 = dust.helpers.tap( params.lat2, chunk, context );
+    var lon2 = dust.helpers.tap( params.lon2, chunk, context );
+    var unit = dust.helpers.tap( params.unit, chunk, context );
+    var decimalPlaces = dust.helpers.tap( params.decimalPlaces, chunk, context ) || 2;
+
+    var radlat1 = Math.PI * lat1/180;
+    var radlat2 = Math.PI * lat2/180;
+    var radlon1 = Math.PI * lon1/180;
+    var radlon2 = Math.PI * lon2/180;
```
Comment (t1gr0u) [inline helpers/calcDistance.js]: call a small function:
```
var degreeToRadian = function( degree ) {
   return Math.PI * degree / 180;
}
```

File: helpers/calcDistance.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    var theta = lon1-lon2;
+    var radtheta = Math.PI * theta/180;
+    var dist = Math.sin( radlat1 ) * Math.sin( radlat2 ) + Math.cos( radlat1 ) * Math.cos( radlat2 ) * Math.cos( radtheta );
+    dist = Math.acos( dist );
+    dist = dist * 180/Math.PI;
+    dist = dist * 60 * 1.1515;
+    if ( unit=="K" ) { dist = dist * 1.609344 }
+    if ( unit=="N" ) { dist = dist * 0.8684 }
```
Comment (taijuten) [inline helpers/calcDistance.js]: Could be useful to have a couple of comments in this section. What do K and N represent? (can work out from the float in the calculation, but good for future readers of the code)

File: package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
 	"name": "the-wall-templater-dustjs",
 	"description": "The Wall compatible template handler",
-	"version": "2.1.2",
+	"version": "2.1.3",
```
Comment (taijuten) [inline package.json]: Should this be a patch, rather than a bugfix?

File: helpers/calcDistance.js (hunk 1 above)
Comment (lukefrake) [inline helpers/calcDistance.js]: Sounds good :+1:

File: helpers/calcDistance.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    var radlon1 = degreeToRadian( lon1 );
+    var radlon2 = degreeToRadian( lon2 );
+    var theta = lon1-lon2;
+    var radtheta = degreeToRadian( theta );
+    var dist = Math.sin( radlat1 ) * Math.sin( radlat2 ) + Math.cos( radlat1 ) * Math.cos( radlat2 ) * Math.cos( radtheta );
+    dist = Math.acos( dist );
+    dist = dist * 180/Math.PI;
+    dist = dist * 60 * 1.1515;
+    
+    // k = kilometeres
```
Comment (lukefrake) [inline helpers/calcDistance.js]: Can you cap these like the units below

**Comment by lukefrake** [comment]
:shipit:

File: helpers/calcDistance.js (hunk 3 above)
Comment (taijuten) [inline helpers/calcDistance.js]: Sorry, just been pointed out that there are actually comments for this. I am a banana.

**Pull request merged**